lib/browser/api/browser-view.ts

```diff
@@ -55,23 +61,39 @@ export default class BrowserView {
 
   // Internal methods
   get ownerWindow (): BrowserWindow | null {
-    return !this.webContents.isDestroyed() ? this.webContents.getOwnerBrowserWindow() : null;
+    return this.#ownerWindow;
   }
 
+  // We can't rely solely on the webContents' owner window because
+  // a webContents can be closed by the user while the BrowserView
+  // remains alive and attached to a BrowserWindow.
   set ownerWindow (w: BrowserWindow | null) {
-    if (this.webContents.isDestroyed()) return;
-    const oldWindow = this.webContents.getOwnerBrowserWindow();
-    if (oldWindow && this.#resizeListener) {
-      oldWindow.off('resize', this.#resizeListener);
+    if (this.#ownerWindow && this.#resizeListener) {
+      this.#ownerWindow.off('resize', this.#resizeListener);
       this.#resizeListener = null;
     }
-    this.webContents._setOwnerWindow(w);
+
+    if (this.webContents && !this.webContents.isDestroyed()) {
+      this.webContents._setOwnerWindow(w);
+    }
+
+    this.#ownerWindow = w;
     if (w) {
       this.#lastWindowSize = w.getBounds();
       w.on('resize', this.#resizeListener = this.#autoResize.bind(this));
+      w.on('closed', () => {
+        this.#ownerWindow = null;
+        this.#destroyListener = null;
```

Comment:
Do we need to re-bind `this.#destroyListener` if
1. w1 is set as the owner window
2. w1 is closed
3. w2 is set as the owner window

Reply:
Good question! At the moment I don't think so, since calling `close()` on a webContents is a one-way street at the moment, and when w1 is closed the BrowserViews' webContents will all be closed: https://github.com/electron/electron/blob/d090f60788b9f61c56782304ef754609b0b0556e/lib/browser/api/browser-window.ts#L80-L82. I could always be wrong though 🤔

Reply:
Ahh right, that makes sense. Thank you for clarifying!